src/Map.tsx

```diff
@@ -44,19 +52,38 @@ const Map: React.FC<MapProps> = ({
         height='100%'
         onViewportChange={setViewport}
         onClick={displayFeature}
+        onHover={handleHover}
+        interactiveLayerIds={
+          Array.isArray(data) && data.length > 0 ? ['layerFill'] : []
+        }
       >
         {data &&
-          data.map((datum, i) => (
-            <Source key={`map=src-${i}`} type='geojson' data={datum}>
-              <Layer
-                id='test'
-                type='fill'
-                paint={{
-                  'fill-color': '#A150F2'
-                }}
-              />
-            </Source>
-          ))}
+          data.map((datum, i) => {
+            const key = `map=src-${i}`;
+            return (
+              <Source id={key} key={key} type='geojson' data={datum}>
+                <Layer
+                  id='layerFill'
+                  type='fill'
+                  paint={{
+                    'fill-color': '#A150F2'
+                  }}
+                />
+              </Source>
+            );
+          })}
+        {hoveredFeature ? (
```

Comment:
Not sure if it would make any practical difference performance-wise, but we don't need to remove the `Source` when there isn't a hovered feature since Mapbox should be able to handle an empty data source on its own.

Reply:
Getting this weird error when I try it.
![image](https://user-images.githubusercontent.com/47393759/127588261-e5e49220-8229-4445-a724-1d87da50d3b3.png)

Reply:
We probably can't pass it `undefined`, but we can give it an empty `FeatureCollection`.

Reply:
The empty `FeatureCollection` is also throwing the same error out but I might have to fill out some of the properties. I'm gonna hold on this tonight and take another look tomorrow.